src/main/java/net/mcreator/workspace/elements/VariableElementType.java

```diff
@@ -22,55 +22,69 @@
 import net.mcreator.generator.mapping.NameMapper;
 import net.mcreator.workspace.Workspace;
 
-public enum VariableElementType {
+import javax.annotation.Nullable;
+import java.awt.*;
 
-	@SerializedName("string") STRING(0x609986), @SerializedName("logic") LOGIC(
-			0x607c99), @SerializedName("number") NUMBER(0x606999), @SerializedName("itemstack") ITEMSTACK(0x996069);
+public class VariableElementType {
 
-	private final int color;
+	//Define each global variables. The instantiation is made in VariableElementTypeLoader.
+	public static VariableElementType STRING;
+	public static VariableElementType LOGIC;
+	public static VariableElementType NUMBER;
+	public static VariableElementType ITEMSTACK;
 
-	VariableElementType(int color) {
-		this.color = color;
+	private String name;
+	private String color;
+	//Use for compatibility with old workspaces

```

Comment:
Could you explain why is this here and what is its purpose? And are all these fields for compatibility or only this one and blockName and how is this used?

Reply:
Fields used for compatibility are only `dependencyType` and `blockName`. The other fields are needed by all variables as they can't use the `type` field. I will change the order to make that more understandable.

Reply:
Please explain what you mean by compatibility and what value do these hold and why we need them as I can't fully understand them

Reply:
Concerning `dependencyType`, in the mappings, you decided to name the dependency used for the LOGIC variables `boolean`, so I couldn't change it for `logic` without breaking workspaces. I decided to add `dependencyType` to solve this problem.
Concerning `blockName`, it's the same problem, except you named all String procedure blocks using `text` instead of String. So, I had to add this new field, so String variables and variable blocks will not break old workspaces.

They have been simply added to fix these two problems because, for all other variables, the `name` field will be used by default.

Do you understand know what I mean by compatibility with old workspaces and why they have been added?

Reply:
I do. But this makes code very hard to read. I suggest avoiding legacy code and adapting workspace format for old workspaces instead. Remove legacy code and update workspaces when reading them.

This will also assure names will be unified from now on to fix my mistake with different names.

Check how I added backwards compatibility when I added multiple sounds support:

1. https://github.com/MCreator/MCreator/blob/master/src/main/java/net/mcreator/workspace/WorkspaceFileManager.java#L48
2. https://github.com/MCreator/MCreator/blob/852200163b48d9d8184953a4755f1e79ede550ba/src/main/java/net/mcreator/workspace/elements/SoundElement.java#L89

Reply:
For the `blockName`, I made a converter as it was inside procedure mod elements and for the dependencyType, I simply changed everything inside the code. When you will have read this message, you can resolve the conversation.

Reply:
Ok. I will do review of the change in the upcoming days when I get a bit larger chunk of time :)